test/issue8/proto.pb.go

```diff
@@ -64,6 +66,11 @@ func (m *Foo) Unmarshal(data []byte) error {
 		}
 		fieldNum := int32(wire >> 3)
 		wireType := int(wire & 0x7)
+
+		if fn := int(fieldNum - 1); fn <= 0 {
```

Comment:
Or we could do even better!
1. count all required fields + get their names
2. generate a struct inside each function with specific bits for each required field
3. generate code specific to the field to set the corresponding bit in that struct
4. generate code that check that all bits are set after parsing is done

This way - the bitmask would grow with the number of required fields, not max field id in the message.

Sorry that i'm just making suggestions here, and showing no code, got carried away :)